### PR TITLE
Add LRU option for in-memory storage

### PR DESCRIPTION
**Pull request by jpage-godaddy**
- Add an `LRU` cache class
- Support using this LRU instead of the simple in-memory cache

### CODE REVIEW

**Review by rmarkins-godaddy — APPROVED** [review]

**Review by agerard-godaddy — APPROVED** [review]

File: lib/lru.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ */
+class LRUCache {
+  /**
+  * @param {Object} [opts]          - Options for the Client instance
+  * @param {Object} [opts.items]    - Items to pre-populate in the cache
+  * @param {number} [opts.maxItems] - Maximum items to retain in the cache
+  * @param {number} [opts.maxAge]   - Age in milliseconds before an item expires
+  */
+  constructor({ items = {}, maxItems, maxAge } = {}) {
+    if (typeof maxItems !== 'number' || maxItems < 1) {
```
Comment (cburton-godaddy) [inline lib/lru.js:14]: **issue (non-blocking):** Does not handle `NaN`

```
> if (typeof NaN !== 'number' || NaN < 1) { console.log('guarded'); } else { console.log('allowed'); }
allowed
```

File: lib/lru.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  /**
+   * Stores a cache item
+   *
+   * @param {String} key    - The cache key
+   * @param {JSONSerializable} value  - The JSON-serializable value to store
+   *
+   * @returns {Promise<void>} a Promise which resolves once storage completes.
+   */
+  async set(key, value) {
+    this._items.set(key, value);
```
Comment (cburton-godaddy) [inline lib/lru.js:57]: **question (non-blocking):** Does this handle any data type or structure robustly? I.e. if I set a cache value that has a circular reference (maybe itself for example), does it handle that?

File: lib/lru.js (hunk 1 above)
Comment (cburton-godaddy) [reply lib/lru.js:14]: `Number.isInteger(NaN)` would help here.

File: lib/lru.js (hunk 2 above)
Comment (jpage-godaddy) [reply lib/lru.js:57]: It just stores the object references; it doesn't serialize before storing.

**Pull request merged**